src/webview/js/js.js

```diff
@@ -904,6 +904,28 @@ documentBody.addEventListener('click', (e: MouseEvent) => {
     return;
   }
 
+  if (target.matches('.poll-vote')) {
+    const messageElement = target.closest('.message');
+    if (messageElement) {
+      // This duplicates some logic from PollData.handle.vote.outbound, but it's
+      // much simpler to just duplicate it than it is to thread a callback all
+      // the way over here.
+      const current_vote = requireAttribute(target, 'data-voted') === 'true';
+      const vote = current_vote ? -1 : 1;
+      sendMessage({
+        type: 'vote',
+        messageId: requireNumericAttribute(messageElement, 'data-msg-id'),
+        key: requireAttribute(target, 'data-key'),
+        vote,
+      });
+      target.setAttribute('data-voted', (!current_vote).toString());
+      target.innerText = (parseInt(target.innerText, 10) + vote).toString();
```

Comment:
Hmm, interesting.

So one consequence of this local-echo strategy is that if for whatever reason the message gets rerendered -- for example because another vote came in, or because some other message changed or a new message arrived in this narrow -- this change will get forgotten. That seems awkward.

What does the webapp do to handle the latency of voting?

Reply:
I'm not sure, @showell, any idea about that?

My inclination is that building a particularly fancy local-echo is not worth it, but I'd be open to suggestions about small tactical fixes.

Reply:
There is no local echo in the webapp. I don't think that's acceptable in the mobile app, where it's easy to think you mistapped if the feedback is delayed.